refactor: move snap services to separate manager & monitor

st3v3nmw opened the pull request:
~Makes sure the extra `services` field doesn't trip up the server after the next landscape server release~

Perfect5th commented:
We need to be a bit careful here. Releasing this client version without backporting the fix to still-supported server versions will break those server versions.

I'd like to suggest a non-backward-breaking option instead: we use a different message type for snap services. Server automatically ignores message types it isn't aware of, and that way reporting installed snaps will still work even if reporting services doesn't.

The new message type can still be sent by `SnapMonitor`, or we could introduce a separate monitor/manager for services, which may give us flexibility if we find that we need service-specific reporting cadence or data handling.

st3v3nmw commented:
I've refactored the code to introduce a separate manager and monitor set for snap services (`SnapServicesManager` & `SnapServicesMonitor`). 

Also fixed the snaps message in `SnapManager._send_snap_update` to match the outgoing message schema. The list of snaps was being attached to snaps instead of `snaps.installed`. This also matches the behavior of `SnapMonitor`

Perfect5th commented on landscape/client/monitor/snapservicesmonitor.py:
```diff
@@ -0,0 +1,27 @@
+
+    def register(self, registry):
+        self.config = registry.config
+        super().register(registry)
```
you may need to set `self.run_interval` for the monitor to be invoked periodically, but I guess until we have a server version that accepts these messages it's kind of moot.

st3v3nmw replied:
Added a 60 seconds `run_interval`.

Also, seems like the monitors inherit a default 5 seconds `run_interval` from `DataWatcher > MonitorPlugin > BrokerClientPlugin.run_interval`

Perfect5th commented on landscape/message_schemas/server_bound.py:
```diff
@@ -830,7 +840,6 @@
                     ),
                 ),
             },
```
To save us in future of something like this happening again, it might be useful to think whether we can anticipate needing to add new keys to any of these `KeyDict`s, and if so, set them to `strict=False`. If there's no reason to do so for any of these, just let me know.

st3v3nmw replied:
I think it's safe to since the `snap-services` message isn't mature yet.

The inner one `services.running` does have `strict=False`, let me add that to `services`

Perfect5th commented:
Also tested against current version of landscape-server. Seems good.
